src/components/preferences/Webhooks.js

```diff
@@ -82,7 +90,6 @@ export default function Webhooks(props) {
                         name="webhook.type.type"
                         id={buildID(baseId, ids.WEBHOOK_TYPES_SELECT)}
                         label={t("type")}
-                        required
```

Comment:
This could actually be set to `true` if there is a value entered for the URL field:
`required={!!(values?.webhook?.url)}`

Reply:
Actually, doing this only partially solved the problem 🤔 . Now the `formik` built in `Required` validation marker shows up but our custom error message still doesn't show until the field is touched.

Reply:
![image](https://user-images.githubusercontent.com/1250790/134531031-60e23d54-249c-4610-9c99-2308dc707ae9.png)

Reply:
After field is touched:
![image](https://user-images.githubusercontent.com/1250790/134531136-caa69baa-c63b-449f-8c7f-33f4c5d01a9f.png)

Reply:
This might be because every field in the form should have an initial value, even if it's an empty string: https://github.com/formium/formik/issues/445#issuecomment-366952762

So the `mapPropsToValues` might need to return an empty webhook.

Looking at these form field names, maybe it needs to return something like this:
```javascript
return {
    ...bootstrap.preferences,
    webhook: { url: "", type: { type: "" } },
}
```

Reply:
Ahh yes. I keep forgetting to check initial values.